docs/content/Configuration/Config.mdx

```diff
@@ -314,6 +315,31 @@ module.exports = {
 };
 ```
 
+### <--{"id" : "Options Reference"}--> canSwitchSqlUser
+
+Used in [SQL API][ref-sql-api], and can be an `async` function. Default
+implementation depends on `CUBEJS_SQL_SUPERUSER` and return `true` when it's equal to session's user.
+
+Called on each change request from Cube SQL API.
+
+For example, you can use `canSwitchSqlUser` to define your custom logic.
+
+```javascript
+module.exports = {
+  canSwitchSqlUser: (current, username) => {
+    if (current === 'admin') {
+        return true;
+    }
+
+    if (current === 'service') {
+        return username !== 'admin';
+    }
+
+    return false;
+  },
+};
```

Comment:
Can we use an `async` function here and return `Promise.resolve`d values instead?

Reply:
Yes, user is allowed to use async function here.

Reply:
Sorry I meant can we change the example here to be a Promise-returning one instead?

Reply:
Move to async fn.